lib/router/route-info.ts

```diff
@@ -17,7 +17,7 @@ export type IModel = {} & {
 
 export type ModelFor<T> = T extends Route<infer V> ? V : never;
 
-export interface Route<T extends IModel = {}> {
+export interface Route<T = unknown> {
```

Comment:
To clarify for lookers-on: because `unknown | undefined` is just `unknown`: `unknown` is a "top" type which means all other types in union with it reduce to it: it already includes all of them, except for `never`.

Reply:
That's true, but then my question is whether, at runtime, it's possible for a (well-typed!) route which *does* exclude `undefined` to call `serialize` with a type *other* than the `T` provided by the route type. If that's possible, then `serialize` itself should simply take `unknown`; if it's not, it can safely take `T` rather than `T | undefined`.

Reply:
My understanding is that serialize is _supposed_ to be called with the current model/context. This would mean that `T | undefined` is correct here. Of course, if this turns out to be a wrong assumption, we should fix it!